simplified-books-controller/src/main/java/org/nypl/simplified/books/controller/BookCoverFetchTask.kt

```diff
@@ -61,7 +76,7 @@ class BookCoverFetchTask(
         this.taskRecorder.finishSuccess(Unit)
       }
     } catch (e: Throwable) {
-      this.logger.error("failed to fetch cover: ", e)
+      this.logger.error("failed to fetch ${this.type}: ", e)
```

Comment:
Not sure if you know, but the SLF4J API allows for this:

```
this.logger.error("failed to fetch {}: ", this.type, e)
```

The `{}` syntax is just positional argument formatting. Additionally, if there are `N` occurrences of `{}` in the string, and the `N + 1`th argument is an exception, then the exception stacktrace is printed as part of the message.

Using `{}` is mildly more efficient because SLF4J lazily evaluates `toString()` on each the arguments, and it won't do it if logging is disable for that logger.

I'm only putting this because you might not know about it. It's bordering on micro-optimization, so it really doesn't matter here.

Reply:
Yep, the reason I didn't use formatting args here is because Logger doesn't seem to have a method overload of `error` that accepts varargs and a Throwable.

I've seen other logging APIs get around this by making the Throwable the first argument so they can have varargs at the end. 

```
public static void e(Throwable t,
                     String message,
                     Object... args)
```

Reply:
Not sure if I'm misunderstanding. All of the SLF4J methods implicitly accept a `Throwable` as the last argument of the `Object...`. 
I'd link to the actual docs, but the SLF4J site isn't responding. This article covers it, though:

https://www.baeldung.com/slf4j-log-exceptions

> In presence of multiple parameters, if the last argument in a logging statement is an exception, then SLF4J will presume that the user wants the last argument to be treated as an exception instead of a simple parameter:

```
logger.error("{}, {}! An exception occurred!", 
  "Hello", 
  "World", 
  new Exception("Custom exception"));
```

> In the above snippet, the String message will be formatted based on the passed object details. We've used curly braces as placeholders for String parameters passed to the method.

> In this case, the output will be:

```
ERROR packageName.NameOfTheClass - Hello, World! An exception occurred!
java.lang.Exception: Custom exception 
  at packageName.NameOfTheClass.methodName(NameOfTheClass.java:lineNo)
```

Reply:
> In presence of multiple parameters, if the last argument in a logging statement is an exception, then SLF4J will presume that the user wants the last argument to be treated as an exception instead of a simple parameter.

Ah! That's unclear from the API. I would have expected it to throw an exception due to the wrong number of arguments. I will refactor.